### DISCUSSION TITLE
Failed to contact car using Tesla API

### DISCUSSION THREAD

**Discussion by alexmereness** [marcone/teslausb#817] [Q&A] [answered]
Hoping to get some help. I noticed that I'm getting very sporadic uploads to network share. I was able to quickly determine the root cause was the Pi losing power from the Tesla. I can open my Tesla app and then the upload process would resume, confirming my suspicions. Looking at the archive log, I can see that TeslaUSB is attempting to hit the Tesla API, but seemingly is failing to connect, resulting in repeated unsuccessful retry attempts (see log below). I have TeslaUSB configured with my refresh token. I only have one car on my Tesla account, but since it wasn't working, I attempted to also include TESLA_NAME and TESLA_VIN. Unfortunately, that made no difference. Any ideas of what might be happening or potential fixes? 

**Relevant TelsaUSB Config:**
export TESLA_REFRESH_TOKEN="<refresh_token>"
export TESLA_NAME="Tesla M3"
export TESLA_VIN="<VIN>"

**Hardware**
-Pi Zero 2W
-128gb Samsung Pro Endurance Micro SD Car
-2018 Tesla Model 3 (AWD-LR)

Archive Server: Windows 10 w/network file share

**Archive Log (Most recent snippet)**

==============================================
Wed  7 Feb 19:30:38 MST 2024: Starting archiveloop at 19.81 seconds uptime...
Wed  7 Feb 19:30:38 MST 2024: Running fsck on /backingfiles/cam_disk.bin...
Wed  7 Feb 19:30:38 MST 2024: | fsck from util-linux 2.33.1
Wed  7 Feb 19:30:40 MST 2024: | fsck.fat 4.1 (2017-01-24)
Wed  7 Feb 19:30:40 MST 2024: | Long filename fragment "2023-11-24_04" found outside a LFN sequence.
Wed  7 Feb 19:30:40 MST 2024: |   (Maybe the start bit is missing on the last fragment)
Wed  7 Feb 19:30:40 MST 2024: |   Not auto-correcting this.
Wed  7 Feb 19:30:40 MST 2024: | Long filename fragment "2023-11-12_10" found outside a LFN sequence.
Wed  7 Feb 19:30:40 MST 2024: |   (Maybe the start bit is missing on the last fragment)
Wed  7 Feb 19:30:40 MST 2024: |   Not auto-correcting this.
Wed  7 Feb 19:30:40 MST 2024: | /dev/loop0p1: 348 files, 600164/1965055 clusters
Wed  7 Feb 19:30:40 MST 2024: Finished fsck on /backingfiles/cam_disk.bin.
Wed  7 Feb 19:30:40 MST 2024: taking snapshot of cam disk in /backingfiles/snapshots/snap-000890
Wed  7 Feb 19:30:41 MST 2024: took snapshot
Wed  7 Feb 19:30:41 MST 2024: comparing new snapshot with /backingfiles/snapshots/snap-000889/snap.bin
Wed  7 Feb 19:30:41 MST 2024: new snapshot is identical to previous one, discarding
Wed  7 Feb 19:30:44 MST 2024: Trying to set time...
Wed  7 Feb 19:31:08 MST 2024: Time adjusted by 23.717918 seconds after 1.130000 seconds
Wed  7 Feb 19:31:08 MST 2024: Starting background task to keep car awake.
Wed  7 Feb 19:31:08 MST 2024: Archiving...
Wed  7 Feb 19:31:08 MST 2024: Ensuring cam archive is mounted...
Wed  7 Feb 19:31:08 MST 2024: Mounting /mnt/archive...
Wed  7 Feb 19:31:09 MST 2024: Mounted /mnt/archive.
Wed  7 Feb 19:31:09 MST 2024: Ensured cam archive is mounted.
Wed  7 Feb 19:31:09 MST 2024: Checking saved folder count...
Wed  7 Feb 19:31:11 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:31:15 MST 2024: There are 17 event folder(s) with 3107 file(s) and 0 track mode file(s) to move.  8 short recording(s) will be skipped.
Wed  7 Feb 19:31:15 MST 2024: Starting recording archiving
Wed  7 Feb 19:31:15 MST 2024: Archiving 3107 file(s) including 17 event folder(s) starting at Wed  7 Feb 19:31:15 MST 2024
Wed  7 Feb 19:31:18 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:31:25 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:31:32 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:31:39 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:31:46 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:31:53 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:32:00 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:32:07 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:32:14 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:32:21 MST 2024: Failed to contact car using Tesla API, retrying in 5s...
Wed  7 Feb 19:32:28 MST 2024: Failed to contact car using Tesla API, retrying in 5s...

**Comment by marcone** [comment]
This should work again, though unclear how long it can remain working before Tesla shuts down the old API.
For my test setup I had to create a new refresh token to get things working again. I used the "Tesla Tokens" Android app to create a new refresh token, and then on the Pi I did:
```
sudo -i
source /root/bin/envsetup.sh
rm /mutable/cache.json /mutable/tesla_api.json
export TESLA_REFRESH_TOKEN='<new token goes here>'
/root/bin/tesla_api.py list_vehicles
```

**Reply by alexmereness** [reply]
Thanks for the reply! I got around to trying this. I followed the steps you took, but when I rean the /root/bin/tesla_api.py list_vehicles command, I received the following error response: 

_Tue 20 Feb 12:15:07 MDT 2024: {
  "response": null,
  "error": "Endpoint is only available on fleetapi. Visit https://developer.tesla.com/docs for more info",
  "error_description": ""
}_

I tried other available API commands via tesla_api.py and was met with the same error response for all the ones I tried. 

![image](https://github.com/marcone/teslausb/assets/40445428/0da9e780-e7b2-4893-97b7-f1f506377b83)

Based on some of your previous comments to others, I know you don't use the awake function in TeslaUSB much, so I really appreciate any guidance or support on this - as well as your overall efforts on the project.

**Reply by marcone** [reply, marked as answer]
That sounds like you didn't [upgrade to the latest TeslaUSB](https://github.com/marcone/teslausb/wiki/FAQ#q-how-do-i-update-to-the-latest-version-of-teslausb) first.

**Reply by alexmereness** [reply]
Thank you, I came to that conclusion as well after doing some digging. I have a question, disclaimer: probably a 'noob' question...part of the reason I didn't initially update was because when I looked at "releases", I saw the last release was "teslausb-20211106.2" from October 2023...but when I look at individual commits and updated files, it's clear things are being updated/changed. I'm trying to understand if/when my TeslaUSB would pull updated files down. For example, I noticed support for Optional Lightshow Drive was added to the teslausb_setup_varibles.conf file.  (https://github.com/marcone/teslausb/blob/main-dev/pi-gen-sources/00-teslausb-tweaks/files/teslausb_setup_variables.conf.sample). I checked my TeslaUSB, both the /root/teslausb_setup_variables_conf and /home/pi/teslausb_setup_variables.conf.sample. Neither file seems to have the updated lightshow configuration, yet I just ran the update today. Just trying to get a better understanding of how this works. I'd like to take advantage of the lightshow and other features as they become available. Once again, many thanks for your efforts.

**Reply by marcone** [reply]
Those releases are for the prebuilt Raspberry Pi OS image that TeslaUSB gets installed on top of. You don't necessarily need to even use that image, and in fact in some cases you can't, like when you're not using a Raspberry Pi.
In any case, when you install or upgrade TeslaUSB, you always get the latest version of TeslaUSB, regardless of the underlying OS.

**Reply by alexmereness** [reply]
Makes sense. My other question was related to the updated teslausb_setup_variables.conf file. I can see the file was updated, on Github, but I'm not seeing the updated on my device. I would like to use the added lightshow drive. What would I need to do to take advantage of that and why would it not be updated on my device?

**Reply by marcone** [reply]
That file isn't updated on the device because that would overwrite the changes you made.
You can just add the LIGHTSHOW_SIZE entry though.
(you'll also need to delete /backingfiles/cam_disk.bin before running `setup-teslausb upgrade`, because otherwise it won't create the new drive)

**Reply by n-gineer** [reply]
> That file isn't updated on the device because that would overwrite the changes you made. You can just add the LIGHTSHOW_SIZE entry though. (you'll also need to delete /backingfiles/cam_disk.bin before running `setup-teslausb upgrade`, because otherwise it won't create the new drive)

Recommend adding to conf.sample with this note so others know how to upgrade with new partitions.
